src/coreclr/jit/optcse.cpp

```diff
@@ -812,7 +812,7 @@ bool Compiler::optValnumCSE_Locate()
                     if (!enableConstCSE &&
                         // Unconditionally allow these constant handles to be CSE'd
                         !tree->IsIconHandle(GTF_ICON_STATIC_HDL) && !tree->IsIconHandle(GTF_ICON_CLASS_HDL) &&
-                        !tree->IsIconHandle(GTF_ICON_STR_HDL))
+                        !tree->IsIconHandle(GTF_ICON_STR_HDL) && !tree->IsIconHandle(GTF_ICON_OBJ_HDL))
```

Comment:
Doesn't `GTF_ICON_STR_HDL` come with an invariant indirection on top? Do you know why we would CSE the constant but not the indir (now and also before)?

Reply:
@jakobbotsch normally it's expected that indirs will be CSE'd, unlike raw constants they have higher weight and CSE should handle them on x64, I don't remember why exactly I put `GTF_ICON_STR_HDL` here in that other PR but AFAIR it affected diffs so I left it (initially I only wanted to handle `GTF_ICON_CLASS_HDL`) - I'll check diffs after this PR what will happen if I remove `GTF_ICON_STR_HDL` from here

Reply:
Ah, good point, will check